File: MyPow.cs
public class Solution {
    public double MyPow(double x, int n) {
     
        // Default
        if (x == 0 || Math.Abs(x) == 1)
        {
            if (n % 2 == 0)
            {
                return Math.Abs(x);    
            }
            else
            {
                return x;
            }
        }
        
        double pow = 1;
        
        if (n > 0)
        {
            for (int i = 0; i < n; i++)
            {
                pow *= x;
            }
        }
        else
        {
            for (int i = 0; i > n; i--)
            {
                pow /= x;
                
                if (pow == 0)
                {
                    return 0;
                }
            }   
        }        
        
        return pow;
        
    }
}

File: Python/myPow.py
class Solution:
    def myPow(self, x: float, n: int) -> float:

        if x == 1:
            return x
        elif x == -1:
            if n % 2 == 0:
                return -x
            else:
                return x

        val = 1

        for i in range(abs(n)):
            if n >= 0:
                val = val * x
            else:
                val = val / x
            if abs(val) < 0.000001:
                    return val

        return val